src/diffpy/structure/structure.py

```diff
@@ -711,8 +711,9 @@ def _get_composition(self):
     element = _linkAtomAttribute(
         "element",
         """Character array of `Atom` types. Assignment updates
-        the element attribute of the respective `Atoms`.""",
-        toarray=numpy.char.array,
+        the element attribute of the respective `Atoms`.
+        Set the maximum length of the element string to 5 characters.""",
+        toarray=lambda items: numpy.char.array(items, itemsize=5),
```

Comment:
A simple solution is to allocate the fixed numpy.char.array length of 5 to begin with.

I chose 5 because while elements have the max str length of 2, I noticed that some have +/- charges for elements/labels.

@sbillinge what do you think? I think this is the main discussion point.

Reply:
Yes, I agree with this solution. 👍